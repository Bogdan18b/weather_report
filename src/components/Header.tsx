import React from "react";
import styled from "styled-components";

const Title = styled.h1`
  text-align: center;
  font-size: 6rem;
  font-weight: bold;
  padding: 1rem;
`;

interface Props {
  city: string;
}

const Header: React.FunctionComponent<Props> = ({ city }) => (
  <Title>{`Weather Report for ${city}`}</Title>
);

export default Header;
