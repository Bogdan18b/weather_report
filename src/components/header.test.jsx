import Header from './header';

it("renders correctly", () => {
  const wrapper = shallow(
    <Header city="Boston" country="US" />
  );
  expect(wrapper).toMatchSnapshot();
});

it("shows city name correctly", () => {
  const wrapper = shallow(
    <Header city="Boston" country="US" />
  );
  const text = wrapper.find("h1").text();
  expect(text).toContain("Boston");
});
