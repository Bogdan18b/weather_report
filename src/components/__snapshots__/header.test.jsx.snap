// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    className="header-title"
  >
    Weather Report for 
    Boston, US
  </h1>
  <h3
    className="header-subtitle"
  >
    Find out the temperature for the next 5 days
  </h3>
</div>
`;
